repose-aggregator/components/services/akka-http-client-service/akka-http-client-service-impl/src/main/java/org/openrepose/core/services/serviceclient/akka/impl/AkkaServiceClientImpl.java

```diff
@@ -103,10 +113,39 @@ public ServiceClientResponse get(String hashKey, String uri, Map<String, String>
 
     @Override
     public ServiceClientResponse get(String hashKey, String uri, Map<String, String> headers, boolean checkCache) throws AkkaServiceClientException {
-        AuthGetRequest authGetRequest = new AuthGetRequest(hashKey, uri, headers);
+
+
         try {
             Timeout timeout = new Timeout(socketTimeout + CONNECTION_TIMEOUT_BUFFER_MILLIS, TimeUnit.MILLISECONDS);
-            Future<ServiceClientResponse> future = getFuture(authGetRequest, timeout, checkCache);
+            Future<ServiceClientResponse> future;
+
+            if (openTracingService.isEnabled()) {
+                // get active span
+                ActiveSpan parentSpan = openTracingService.getGlobalTracer().activeSpan();
+
+                try (ActiveSpan span = openTracingService.getGlobalTracer().buildSpan(uri)
+                    .asChildOf(parentSpan)
+                    .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT)
+                    .withTag(Tags.HTTP_URL.getKey(), uri)
+                    .withTag(Tags.HTTP_METHOD.getKey(), "GET")
+                    .startActive()) {
+
+                    // this needs to be better ... however, it looks like headers are an immutable map
+                    Map<String, String> copiedHeaderMap = new HashMap<>(headers.size() + 1);
+                    copiedHeaderMap.putAll(headers);
+
+                    // adds the trace guid (tracer specific)
+                    openTracingService.getGlobalTracer().inject(
+                        span.context(), Format.Builtin.TEXT_MAP,
+                        new TraceGUIDInjector(copiedHeaderMap, CommonHttpHeader.TRACE_GUID));
```

Comment:
Side effects on a passed in parameter is bad business. Why are you passing the whole map and modifying it? Do you use anything other than the one value? If not I'd ay pass the value you care about and and have it return the new value.